test/tools/yulopti.cpp

```diff
@@ -90,33 +91,29 @@ class YulOpti
 	{
 		ErrorList errors;
 		ErrorReporter errorReporter(errors);
-		CharStream _charStream(_input, "");
+		CharStream charStream(_input, "");
+
 		try
 		{
-			m_ast = yul::Parser(errorReporter, m_dialect).parse(_charStream);
-			if (!m_ast || !errorReporter.errors().empty())
+			ObjectParser parser(errorReporter, m_dialect);
+
+			auto scanner = make_shared<Scanner>(charStream);
+			m_inputIsCodeBlock = (scanner->currentToken() == Token::LBrace);
+			m_object = parser.parse(scanner, false);
+
+			if (!m_object || !errorReporter.errors().empty())
 			{
 				cerr << "Error parsing source." << endl;
-				printErrors(_charStream, errors);
-				throw std::runtime_error("Could not parse source.");
-			}
-			m_analysisInfo = make_unique<yul::AsmAnalysisInfo>();
-			AsmAnalyzer analyzer(
-				*m_analysisInfo,
-				errorReporter,
-				m_dialect
-			);
-			if (!analyzer.analyze(*m_ast) || !errorReporter.errors().empty())
-			{
-				cerr << "Error analyzing source." << endl;
-				printErrors(_charStream, errors);
-				throw std::runtime_error("Could not analyze source.");
+				printErrors(charStream, errors);
+				solThrow(Exception, "Could not parse source.");
```

Comment:
The exception goes from here to the `catch` block, which will print the errors again. We should print them only once.

Also, `runCodeAnalyzer()` should use `solThrow()` too instead of asserting that it expects no errors (which is not true).